api/v1beta1/azuremanagedcontrolplane_types.go

```diff
@@ -110,8 +110,10 @@ type AzureManagedControlPlaneSpec struct {
 	OutboundType *ManagedControlPlaneOutboundType `json:"outboundType,omitempty"`
 
 	// SSHPublicKey is a string literal containing an ssh public key base64 encoded.
+	// Use empty string to autogenerate new key. Use null value to not set key.
 	// Immutable.
-	SSHPublicKey string `json:"sshPublicKey"`
+	// +optional
+	SSHPublicKey *string `json:"sshPublicKey"`
```

Comment:
Shouldn't this be `json:"sshPublicKey,omitempty"`?

Reply:
Good catch, it should be.